### DISCUSSION TITLE
Clock Speed

### DISCUSSION THREAD

**Discussion by UkrainianBanderasCat** [UkrainianBanderasCat/Digital-Logic-Sim2#9] [General]
We need to add a slider on the clock to control its speed.

**Comment by firecerne** [comment]
We should also be able to activate or deactivate it , because , for some reason , components generating a signal without any inputs don't work inside chips.

**Reply by UkrainianBanderasCat** [reply]
In debug mode, you can control it

**Reply by firecerne** [reply]
Yes, but the problem is that the chip doesn't work inside other chips. It is possible to either make the clock toggleable , or change other more complex parts of the code.

**Comment by AlphaCodingPilot** [comment]
I Added an option to change the clock speed outside of debug mode.
The reason the clock does'nt work inside other chips is that custom chips only evaluate they're inputs and since the clock does not have any input pins it does'nt get updated. I'm currently working on fixing this issue.
